functions

```diff
@@ -29,10 +29,7 @@ emergency_shell() {
 
 detect_virt() {
    # Detect LXC containers
-   [ ! -e /proc/self/environ ] && return
-   if grep -q lxc /proc/self/environ >/dev/null; then
```

Comment:
For the record, this is looking for the environment variable declaration:

```text
container=lxc
```

Reply:
yes, uda, I understand